gpu/src/BindGroupLayout.zig

```diff
@@ -38,10 +40,72 @@ pub const Entry = extern struct {
     reserved: ?*anyopaque = null,
     binding: u32,
     visibility: ShaderStage,
-    buffer: Buffer.BindingLayout,
-    sampler: Sampler.BindingLayout,
-    texture: Texture.BindingLayout,
-    storage_texture: StorageTextureBindingLayout,
+    buffer: Buffer.BindingLayout = .{},
+    sampler: Sampler.BindingLayout = .{},
+    texture: Texture.BindingLayout = .{},
+    storage_texture: StorageTextureBindingLayout = .{},
```

Comment:
I actually think these should be optional values, according to the spec, not default to zero values.

<img width="539" alt="image" src="https://user-images.githubusercontent.com/3173176/162637808-1a044a9a-4cad-4ec1-b198-4fcc112f9708.png">

Reply:
`BindingLayoutGroup.Entry' is an `extern struct` - we can't have optional field for ABI compatibility.

Reply:
Ahh, you're right!